ice-adapter/src/main/java/com/faforever/iceadapter/IceAdapter.java

```diff
@@ -154,51 +128,52 @@ public synchronized static void onFAShutdown() {
     /**
      * Stop the ICE adapter
      */
-    public static void close() {
+    @Override
+    public void close() {
         log.info("close() - stopping the adapter");
 
         Executor.executeDelayed(500, () -> System.exit(0));
```

Comment:
If make normal "Closeable" objects, they will always be closed. And the application will close normally on its own.

Reply:
I am not sure what you suggest to do here?

This is a remote close call from the other side of the RPC connection (=FAF client). The ice adapter shall run until close is called, because we don't know when the game was closed.